runner/scheduler.py

```diff
@@ -387,6 +387,8 @@ def cost_callback(cost_usd: float):
             example.dataset_generator.cost_callback = cost_callback
 
     def run(self):
+        self.master_log = MasterLog(self.master_log_path)
+        self.runstats_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
This line could be moved to `save_runstats`

Reply:
No, I had to move it from there. The master log saving needs that path too and it runs before the runstats saving does.